src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/scaleset/AzureScaleSetProvider.java

```diff
@@ -0,0 +1,327 @@
+        VirtualMachineScaleSet.DefinitionStages.WithLinuxCreateManagedOrUnmanaged virtualMachineScaleSet_stage2;
+
+        virtualMachineScaleSet_stage2 = vmAdminSSHPubKey.map(virtualMachineScaleSet_stage1::withSsh)
+                                                        .orElseGet(() -> virtualMachineScaleSet_stage1.withRootPassword(vmAdminPassword));
+
+        VirtualMachineScaleSet virtualMachineScaleSet = virtualMachineScaleSet_stage2
+
+                                                                                     .withNewDataDisk(azureVMDiskSize)
+                                                                                     .withCapacity(vmssNbOfInstances)
+
+                                                                                     .defineNewExtension("CustomScriptForLinux")
+                                                                                     .withPublisher("Microsoft.OSTCExtensions")
+                                                                                     .withType("CustomScriptForLinux")
+                                                                                     .withVersion("1.4")
+                                                                                     .withMinorVersionAutoUpgrade()
+                                                                                     .withPublicSetting("fileUris",
+                                                                                                        fileUris)
+                                                                                     .withPublicSetting("commandToExecute",
+                                                                                                        installCommand)
+
+                                                                                     .attach()
+                                                                                     .create();
+
+        logger.info("Azure Scale set '" + azureScaleSetName + "'" + " created inside resource group " + resourceGroup);
+
+        // Return the list of VMs of the Scale Set
+        return virtualMachineScaleSet.virtualMachines()
+                                     .list()
+                                     .stream()
+                                     .map(vm -> instance.withTag(vm.name())
+                                                        .withId(vm.id())
+                                                        .withNumber(SINGLE_INSTANCE_NUMBER))
+                                     .collect(Collectors.toSet());
+    }
+
+    private VirtualMachineScaleSetSkuTypes selectScaleSetSku(String instanceType) {
+        if (instanceType == null)
+            return VirtualMachineScaleSetSkuTypes.STANDARD_D1_V2;
+        if (instanceType.contains("STANDARD_D1_V2"))
+            return VirtualMachineScaleSetSkuTypes.STANDARD_D1_V2;
+        //TODO: add support for more SKU
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1135)